HIP/hip-859.md

```diff
@@ -0,0 +1,158 @@
+---
+hip: 859
+title: Add support for gasConsumed field in REST APIs related to contract results
+author: Ivan Kavaldzhiev <ivan.kavaldzhiev@limechain.tech>, Stoyan Panayotov <stoyan.panayotov@limechain.tech>
+working-group: Steven Sheehy <steven.sheehy@swirldslabs.com>, Nana Essilfie-Conduah <nana@swirldslabs.com>, Danno Ferrin <danno.ferrin@swirldslabs.com>, David Bakin <david.bakin@swirldslabs.com>
+type: Standards Track
+category: Mirror
+needs-council-approval: Yes
+status: Draft
+created: 17.01.2024
+discussions-to: https://github.com/hashgraph/hedera-improvement-proposal/discussions/860
+---
+
+## Abstract
+
+Currently, consensus nodes export a `gasUsed` field for each EVM impacting transaction, as part of the record streaming. As a result, the `contract_result` table in the mirror node database contains a `gas_used` field for each transaction. This field value is calculated using the gas refund policy applied in the consensus node.
+This means that the value in this field represents how much gas the user was actually charged, not the amount of gas that the EVM consumed during the execution. This value can greatly differ from the actual amount that was used by the EVM. 
+
+Let's say we use a 20% refund policy for the `gasLimit` that is sent by the user.
+
+That means if the EVM consumes for example 55_500 gas, but the user is passing 400_000 gas limit, they will be charged the `max(55_500, 320_000)`, so it would be 320_000. That is the value calculated after we return the 20% of the gasLimit back to the user, which is 80_000.
+The logic compares the bigger of the two values - the gas consumed by the EVM and the 80% of the `gasLimit` that the user has passed.
+
+In this way we lose track of the actual gas that was consumed by the EVM.
+
+This HIP will add a new field `gas_consumed` to the `contract_result` table in the mirror node, which will contain the actually consumed gas by the EVM.
+
+```
+create table if not exists contract_result
```

Comment:
The mirror node database is not part of its public API and could change at anytime without another HIP, making this document become inaccurate. We should omit any reference to a specific table or column from this HIP and instead speak in general terms.

Reply:
That's fair.
I believe this was my suggestion for insights into product changes.
Given it's not an external API it's more of a suggestion as part of the HIP so it's fine if the implementation is not exact.
If it's easier and better for accuracy though let's remove it to avoid confusion